src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Bignum.cs

```diff
@@ -16,22 +16,22 @@ internal static partial class Crypto
         internal static extern void BigNumDestroy(IntPtr a);
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_BigNumFromBinary")]
-        private static extern IntPtr BigNumFromBinary(byte[] s, int len);
+        private static extern IntPtr BigNumFromBinary(ref byte s, int len);
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_BigNumToBinary")]
         private static extern unsafe int BigNumToBinary(SafeBignumHandle a, byte* to);
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_GetBigNumBytes")]
         private static extern int GetBigNumBytes(SafeBignumHandle a);
 
-        private static IntPtr CreateBignumPtr(byte[] bigEndianValue)
+        private static IntPtr CreateBignumPtr(ReadOnlySpan<byte> bigEndianValue)
         {
-            if (bigEndianValue == null)
+            if (bigEndianValue.IsEmpty)
```

Comment:
This appears to be valid a valid change for all uses of `CreateBignum`. I did not see any places where `null` could be passed in, in fact `DefaultExponent` in `RSAOpenSsl` seemed to be the only usage.

Reply:
Technically it makes this different from BN_bin2bn.  At least, 1.1.1 seems to turn len==0 => BIGNUM(0).

So it's probably OK to remove the check altogether; but if BN_bin2bn hasn't always had safe-dereferencing semantics then something like this would work:

```C#
byte zero = 0;

if (bigEndianValue.IsEmpty)
{
    bigEndianValue = MemoryMarshal.CreateReadOnlySpan(ref zero, 1);
}
```

Or, since there's only one caller, just `Debug.Assert(!bigEndianValue.IsEmpty);` with a comment that while it might work, it hasn't been verified across all versions on all supported OSes, and it's not an expected scenario.

Reply:
FWIW, I jumped back to 0.9.6 and BN_bin2bn has a check for length == 0 to exit early... so it's probably fine to just remove this if.  MemoryMarshal.GetReference will do the scary "don't read this ref" when given `default(ReadOnlySpan<byte>)`, but that turns into `(void*)0` in the marshaller, so everything should just work.

Reply:
Ah nuts, you beat me to it. I looked in 1.0.2. https://github.com/openssl/openssl/blob/OpenSSL_1_0_2u/crypto/bn/bn_lib.c#L632-L635